cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlClientTest.java

```diff
@@ -0,0 +1,151 @@
+            context.verify(() -> assertThat(result.getJson().containsKey("goalSummary"), is(true)));
+            context.verify(() -> assertThat(result.getJson().containsKey("proposals"), is(true)));
+            context.verify(() -> assertThat(result.getJson().containsKey("loadAfterOptimization"), is(true)));
+            context.verify(() -> assertThat(result.getJson().containsKey("loadBeforeOptimization"), is(true)));
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void testCCRebalanceNotEnoughValidWindowsException(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+
+        MockCruiseControl.setupCCRebalanceResponse(ccServer);
+
+        RebalanceOptions rbOptions = new RebalanceOptions.RebalanceOptionsBuilder().build();
+
+        CruiseControlApiImpl client = new CruiseControlApiImpl(vertx);
+
+        client.rebalance(HOST, PORT, rbOptions, MockCruiseControl.REBALANCE_NOT_ENOUGH_VALID_WINDOWS_ERROR)
+                .setHandler(result -> {
+                    if (result.succeeded()) {
+                        assertThat(result.result().thereIsNotEnoughDataForProposal(), is(true));
+                        context.completeNow();
+                    } else {
+                        context.failNow(result.cause());
+                    }
+                });
+    }
+
+    @Test
+    public void testCCGetRebalanceUserTask(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+
+        MockCruiseControl.setupCCUserTasksResponse(ccServer, 0);
+
+        CruiseControlApi client = new CruiseControlApiImpl(vertx);
+        String userTaskID = MockCruiseControl.REBALANCE_NO_GOALS_RESPONSE_UTID;
+
+        client.getUserTaskStatus(HOST, PORT, userTaskID).setHandler(context.succeeding(result -> {
+            context.verify(() -> assertThat(result.getUserTaskId(), is(MockCruiseControl.USER_TASK_REBALANCE_NO_GOALS_RESPONSE_UTID)));
+            context.verify(() -> assertThat(result.getJson().getJsonObject(CC_REST_API_SUMMARY), is(notNullValue())));
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void testCCGetRebalanceVerboseUserTask(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+
+        MockCruiseControl.setupCCUserTasksResponse(ccServer, 0);
+
+        CruiseControlApi client = new CruiseControlApiImpl(vertx);
+        String userTaskID = MockCruiseControl.REBALANCE_NO_GOALS_VERBOSE_RESPONSE_UTID;
+
+        client.getUserTaskStatus(HOST, PORT, userTaskID).setHandler(context.succeeding(result -> {
+            context.verify(() -> assertThat(result.getUserTaskId(), is(MockCruiseControl.USER_TASK_REBALANCE_NO_GOALS_VERBOSE_RESPONSE_UTID)));
+            context.verify(() -> assertThat(result.getJson().getJsonObject(CC_REST_API_SUMMARY), is(notNullValue())));
+            context.completeNow();
```

Comment:
I generally prefer calling context.checkpoint() outside of the sync call and calling flag() inside this handler